### PR TITLE
fix: 5xx handling for cd-pipeline patch api

### PR DESCRIPTION
**Pull request by adi6859**
# Description
5xx handling for cd-pipeline patch api

Fixes #https://github.com/devtron-labs/devtron/issues/4482

## How Has This Been Tested?
test are performed locally 

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/service/HelmAppServiceWrapper.go [hunk 1]
```diff
@@ -166,10 +168,19 @@ func (impl *ApplicationServiceServerImpl) GetDesiredManifest(ctx context.Context
...
 	if err != nil {
+		isReleaseInstalled, releaseErr := impl.HelmAppService.IsReleaseInstalled(context.Background(), in)
+		if releaseErr != nil {
+			impl.Logger.Errorw("error in checking if release is installed or not")
+			return nil, status.Error(codes.Internal, releaseErr.Error())
+		}
+		if !isReleaseInstalled {
+			impl.Logger.Errorw("error, no release found", "ReleaseIdentifier", in)
+			return nil, status.Error(codes.NotFound, fmt.Sprintf(" release not found for '%s'", in.ReleaseName))
+		}
```
Comment (prakash100198) [inline pkg/service/HelmAppServiceWrapper.go:181]: pls put comments here as to why these changes were necessary

Comment (adi6859) [reply pkg/service/HelmAppServiceWrapper.go:182]: done

**Review by prakash100198 — APPROVED** [review]
LGTM

**Review by nishant-d — APPROVED** [review]
